### ISSUE TITLE
Is there any sample code using this plugin available?

### ISSUE THREAD

**Issue by mts749** [fix/grails-alfresco-plugin#2]
Pointers to sample code using this plugin would be greatly appreciated.

Thanks,
Martin

**Comment by fix** [comment]
Not really here, but in the readme i think this is clear enough:
- register an alfresco server at `http://myserver:8080/myapp/alfrescoServer`
- create a view with a `<alfresco:spaceTree .... />` tag inside

That's it!

 https://github.com/fix/grails-alfresco-plugin#readme

**Comment by mnavas** [comment]
I'm trying to use the library and I get an exception: Cannot get property 'ticket' on null object.
What do you mean with create a connector user/password? Just create user and passwords in alfresco?. I did that and since I sent only username, servername and node, it threw the exception.

We need a more understandable example, also for the configuration of the alfresco side.

Please help!!!

**Comment by fix** [comment]
@mnavas if you can't get a ticket on null object it means that your method cannot find the AlfrescoServer linked to the the username/servername provided in your tag. Be sure when you use `<alfresco:spaceTree username="registered_username" servername="registered_name" node="workspace://StoresSpace/your-node">` you have a registered AlfrescoServer with `name="registered_name"` and `username="registered_username"`.

If you wonder where to register your `AlfrescoServer`, or get the list of `AlfrescoServer` go to `http://your_host_name_probably_localhost:port_probably_8080/yourappname/alfrescoServer`

**Comment by fix** [comment]
but taking your remarks, i'll update the README accordingly

**Comment by mnavas** [comment]
Thanks for the quick response, we installed the plugin but we don't get anything for:

http://localhost:8080/grailsApp/alfrescoServer

From the readme file, we understood that the plugin would create a url in the grails application to register the available alfresco servers. However the controler does not exist and we get the url exception response of apache.

Some examples would be appreciated. 

Thanks

**Comment by fix** [comment]
yes actually there should be something actually! if you inspect your database do you have an "alfresco_server" table?

also, if you create an AlfrescoServer inside your Bootstrap, can it solve the issue?

creating from BootStrap:

```
new org.grails.plugin.alfresco.AlfrescoServer(
    name:"test", 
    details:"test", 
    serverUrl:"http://afrescoServerUrl/",  
    username:"username",
    password:"pwd", 
    ticket:""
).save(failOnError:true)
```

**Comment by kienvm** [comment]
Hi i got this error, please help me. thank you so much
Exception Message: invalid stream header: 4C808A11
Caused by: Error processing GroovyPageView: Error executing tag alfresco:spaceTree: null at D:/New folder/grails-alfresco-plugin-master/grails-app/views/helloThere/helloAlfresco.gsp:10

**Comment by fix** [comment]
Hi the error does not say much. But my feeling is that you forgot to pass an arg in the gsp tag alfresco:spaceTree/
—

**Comment by kienvm** [comment]
Hi, this is my arg in tag <alfresco:spaceTree   
 username="${alfrescoServer?.username }" servername="${alfrescoServer?.name }" node="http://127.0.0.1:8080/alfresco/faces/jsp/admin/store-browser.jsp">  
when load this page i got this error. I think somethings wrong in arg
Exception Message: invalid stream header: 4C808A11

**Comment by fix** [comment]
I think, you should put a different argument for node, this is not the server but the node name, something like: "1a235d-4342d-a8769-9890a"

**Comment by fix** [comment]
but if you just want to browse the tree use alfresco:homeTree tag

**Comment by sharmakuldeep** [comment]
when i try to using alfresco:homeTree tag there is no error or exception but it asks for the user attribute in the tag so what will be the value of this attribute, and also in alfresco:spaceTree/ tag from where i can found the value of node attribute please specify

**Comment by sharmakuldeep** [comment]
Also i have only serverUrl, username and password, can anybody please specify what else i need to connect to the Alfresco server from within my application.

**Comment by sharmakuldeep** [comment]
new org.grails.plugin.alfresco.AlfrescoServer(
    name:"test", 
    details:"test", 
    serverUrl:"http://afrescoServerUrl/",  
    username:"username",
    password:"pwd", 
    ticket:""
).save(failOnError:true)    when i try to save the AlfrescoServer from bootstrap, its not saving the server in database because of the null value of the ticket so what value of the ticket i should provide here.

**Comment by fix** [comment]
sorry to be back so late. What do you mean? ticket is not null in this case, it is setup with empty string.

**Comment by idiengaid** [comment]
hello very interesting plugin so when i'm trying to use it but i'm facing trouble an error message 

Caused by AuthenticationFailure: com.rivetlogic.core.cma.exception.CmaRuntimeException: <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
   <head>
      <title>Web Script Status 404 - Not Found</title>
      <link rel="stylesheet" href="/alfresco/css/webscripts.css" type="text/css" />
   </head>
   <body>
      <div>
         <table>
            <tr>
               <td><img src="/alfresco/images/logo/AlfrescoLogo32.png" alt="Alfresco" /></td>
               <td><span class="title">Web Script Status 404 - Not Found</span></td>
            </tr>
         </table>
         <br/>
         <table>
            <tr><td>The Web Script <a href="%2Falfresco%2Fservice%2Fcma%2Fauthenticationservice">/alfresco/service/cma/authenticationservice</a> has responded with a status of 404 - Not Found.</td></tr>
         </table>
         <br/>
         <table>
            <tr><td><b>404 Description:</b></td><td> Requested resource is not available.</td></tr>
            <tr><td>&nbsp;</td></tr>
            <tr><td><b>Message:</b></td><td>01070044 Script url /cma/authenticationservice does not map to a Web Script.</td></tr>
            <tr><td><b>Server</b>:</td><td>Community v5.2.0 (re21f2be5-b22) schema 10 057</td></tr>
            <tr><td><b>Time</b>:</td><td>7 févr. 2018 18:53:15</td></tr>
            <tr><td></td><td>&nbsp;</td></tr>
         </table>
      </div>
   </body>
</html>
